File: Java/dependency-injection/control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Outbound/MessageOutbound.cs
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Interfaces;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Outbound
{
  public interface IOutbound
  {
    void Send(AcmeMessage message);
  }

  public class MessageOutbound : IOutbound
  {
    private readonly ISocket _outputSocket;
    private readonly IOutboundMessageFactory _outboundMessageFactory;

    public MessageOutbound(
      ISocket outputSocket, 
      IOutboundMessageFactory outboundMessageFactory)
    {
      _outputSocket = outputSocket;
      _outboundMessageFactory = outboundMessageFactory;
    }

    public void Send(AcmeMessage message)
    {
      var outboundMessage = _outboundMessageFactory.CreateOutboundMessage();
      message.WriteTo(outboundMessage);
      outboundMessage.SendVia(_outputSocket);
    }
  }
}

File: Java/dependency-injection/control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Outbound/MessageOutbound.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Outbound;

import com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces.AcmeMessage;
import com.github.grzesiek_galezowski.DependencyInjectionAfter.Outbound.IOutbound;

public class MessageOutbound implements IOutbound {
  private final ISocket _outputSocket;
  private final IOutboundMessageFactory _outboundMessageFactory;

  public MessageOutbound(
      ISocket outputSocket,
      IOutboundMessageFactory outboundMessageFactory) {
    _outputSocket = outputSocket;
    _outboundMessageFactory = outboundMessageFactory;
  }

  public void Send(AcmeMessage message) {
    IOutboundMessage outboundMessage = _outboundMessageFactory.CreateOutboundMessage();
    message.WriteTo(outboundMessage);
    outboundMessage.SendVia(_outputSocket);
  }
}
